pkg/mermaid/datamodeldiagram/datamodeldiagramwithappandtype.go

```diff
@@ -0,0 +1,81 @@
+package datamodeldiagram
+
+import (
+	"fmt"
+
+	"github.com/anz-bank/sysl/pkg/mermaid"
+	"github.com/anz-bank/sysl/pkg/sysl"
+	"github.com/anz-bank/sysl/pkg/syslwrapper"
+)
+
+func GenerateDataModelDiagramWithAppAndType(m *sysl.Module, appName string, typeName string) (string, error) {
+	mapper := syslwrapper.MakeAppMapper(m)
+	mapper.IndexTypes()
+	mapper.ConvertTypes()
+	return generateDataModelDiagramWithAppAndTypeHelper(mapper.SimpleTypes, appName, typeName,
+		&[]externalLink{}, &[]string{}, &[]externalLink{})
+}
+
+func generateDataModelDiagramWithAppAndTypeHelper(simpleTypes map[string]*syslwrapper.Type, appName string,
+	typeName string, externalLinks *[]externalLink, appTypes *[]string, eLinks *[]externalLink) (string, error) {
+	var result string
+	appType1 := fmt.Sprintf("%s:%s", appName, typeName)
+	result = "%% AUTOGENERATED CODE -- DO NOT EDIT!\n\nclassDiagram\n"
+	for appType, value := range simpleTypes {
+		switch value.Type {
+		case relation, tuple:
+			printProperties(value.Properties, appType, externalLinks)
+		case enum:
+			printEnum(value.Enum)
+		case mapS:
+			printMap(value.Properties, appType, value.PrimaryKey, externalLinks)
+		default:
+			panic("Unrecognised type")
+		}
+	}
+	getRelatedTypes(appType1, *externalLinks, appTypes, eLinks)
+	for _, appType := range *appTypes {
+		typeValue := simpleTypes[appType]
+		result += fmt.Sprintf(" class %s {\n", mermaid.CleanString(appType))
+		switch typeValue.Type {
+		case relation, tuple:
+			result += printProperties(typeValue.Properties, appType, externalLinks)
+		case enum:
+			result += printEnum(typeValue.Enum)
+		case mapS:
+			result += printMap(typeValue.Properties, appType, typeValue.PrimaryKey, externalLinks)
+		default:
+			result += ""
```

Comment:
Maybe logging a warning here would be better? `result +=""` doesn't do anything right?

Reply:
Yes! I will added a logger once I get the initial implementation working on catalog. I have a tracker for that issue.

Reply:
👍